docs/default.nix

```diff
@@ -32,74 +32,447 @@ let
+  #     readme = "docs/src/options/platforms/«platform».md";
+  #     defaultReadme = "note about the path above not existing";
+  #     optionsByPlatform.«platform» = [ ... ];
+  #   };
+  # }
+  #
+  # Options are inserted one at a time into the appropriate page, creating
+  # new page entries if they don't exist.
+
+  insert =
+    {
+      index,
+      page,
+      emptyPage,
+      platform,
+      option,
+    }:
+    index
+    // {
+      ${page} =
         let
-          commit = inputs.self.rev or "master";
-          declarationPrefix = toString inputs.self;
+          oldPage = index.${page} or emptyPage;
         in
-        option:
-        option
+        oldPage
         // {
-          declarations = map (
-            declaration:
-            let
-              declarationString = toString declaration;
-              declarationWithoutprefix = lib.removePrefix "${declarationPrefix}/" declarationString;
-            in
-            lib.throwIfNot (lib.hasPrefix declarationPrefix declarationString)
-              "declaration not in ${declarationPrefix}: ${declarationString}"
-              {
-                name = "<${declarationWithoutprefix}>";
-                url = "https://github.com/danth/stylix/blob/${commit}/${declarationWithoutprefix}";
-              }
-          ) option.declarations;
-
-          visible = option.visible && lib.any pathFilter option.declarations;
+          optionsByPlatform = oldPage.optionsByPlatform // {
+            ${platform} = oldPage.optionsByPlatform.${platform} ++ [ option ];
+          };
         };
-    }).optionsCommonMark;
-
-  # The documentation for options which aren't linked to a specific module
-  makePlatformsOptionsDoc =
-    configuration:
-    makeOptionsDoc {
-      inherit configuration;
-      pathFilter =
-        path:
-        lib.hasPrefix "${inputs.self}/" path
-        && !lib.hasPrefix "${inputs.self}/modules/" path;
     };
 
-  # Returns an attribute set of module names and their corresponding option
-  # documentation.
-  makeModuleOptionsDoc =
-    configuration:
-    lib.mapAttrs (
-      module: _:
-      makeOptionsDoc {
-        inherit configuration;
-        pathFilter = lib.hasPrefix "${inputs.self}/modules/${module}/";
+  insertDeclaration =
+    {
+      index,
+      declaration,
+      platform,
+      option,
+    }:
+    # Only include options which are declared by a module within Stylix.
+    if lib.hasPrefix "${inputs.self}/" declaration then
+      let
+        path = lib.removePrefix "${inputs.self}/" declaration;
+        pathComponents = lib.splitString "/" path;
+      in
+      # Options declared in the modules directory go to the Modules section,
+      # otherwise they're assumed to be shared between modules, and go to the
+      # Platforms section.
+      if builtins.elemAt pathComponents 0 == "modules" then
+        let
+          module = builtins.unsafeDiscardStringContext (builtins.elemAt pathComponents 1);
```

Comment:
Why do we bother removing the string context?

Reply:
`module` is used in `page`:

https://github.com/danth/stylix/blob/0ebd2a77b6955c85a1f7973d79e007db599f07fe/docs/default.nix#L114

Which eventually becomes an attribute name:

https://github.com/danth/stylix/blob/0ebd2a77b6955c85a1f7973d79e007db599f07fe/docs/default.nix#L80

And attribute names are not allowed to have context.

For reference, the context is `${inputs.self}`, which has been stripped out of the string here:

https://github.com/danth/stylix/blob/0ebd2a77b6955c85a1f7973d79e007db599f07fe/docs/default.nix#L102

So this is safe to do.

Reply:
Yes, that seems sensible.